PrivateHeaders/APPX/ZIP.h

```diff
@@ -327,6 +328,11 @@ namespace appx {
            << "xmlns=\"http://schemas.microsoft.com/appx/2010/blockmap\" "
            << "HashMethod=\"http://www.w3.org/2001/04/xmlenc#sha256\">";
         for (const ZIPFileEntry &entry : otherEntries) {
+            const std::string suffix = ".appx";
+            if (suffix.size() < entry.fileName.size() &&
+                    std::equal(suffix.rbegin(), suffix.rend(), entry.fileName.rbegin())) {
+                continue;
+            }
```

Comment:
Something else I noticed: we should only exclude .appx files when making bundles, and not when making packages. This code needs to be conditional.

> I can double check but I would rather do what MakeAppx.exe does.

That's fair. Do the .appx files still get hashed into the AXPC section? In other words, does your current set of changes produce .appxbundle-s with valid signatures?

> Also, otherEntries variable in that function is zipFileEntries variable in WriteAppx function in APPX.xpp: it is a list of all zip entries that are basically used before and after the function call. Creating a new copy of that list that only excludes .appx files and is used only for this function doesn't seem like a perfect idea honestly. Are there any strong reasons to do this?

I agree that making a new vector is kinda lame. However, WriteAppxBlockMapZIPFileEntry shouldn't arbitrary ignore files it's given. That doesn't make sense to me. WriteAppxBlockMapZIPFileEntry should make a block map entry for every file it's given.

WriteAppxBlockMapZIPFileEntry could accept an iterator pair. We'd then put this logic in an iterator. Seems an overly-complicated way of avoiding a few string copies though. =S